server/author/views.py

```diff
@@ -58,3 +58,16 @@ def get_object(self):
             raise AuthenticationFailed(detail ={"error": ["User has not been approved by admin"]})
             
         return self.request.user
+
+class AllAuthorsView(generics.ListAPIView):
+    """Get all authors in the system"""
+    serializer_class = AuthorProfileSerializer
+    authenticate_classes = (authentication.TokenAuthentication,)
+    permission_classes = (permissions.IsAuthenticated,)
+    
+    def get_queryset(self):
+        if not self.request.user.adminApproval:
+            raise AuthenticationFailed(
+                detail={"error": ["User has not been approved by admin"]})
+        
+        return get_user_model().objects.filter(type='author')
```

Comment:
Does this get all authors? Including unapproved authors?
Maybe we should check for approved authors

Reply:
this exclude superuser, hmmm im not sure if we want to filter out user without admin approval but I cool with either way.